src/Admin/AjaxAction.php

```diff
@@ -0,0 +1,225 @@
+<?php
+/**
+ * Class AjaxAction.
+ *
+ * @package AmpProject\AmpWP
+ */
+
+namespace AmpProject\AmpWP\Admin;
+
+/**
+ * Base class to define a new AJAX action.
+ *
+ * @package AmpProject\AmpWP
+ */
+final class AjaxAction {
```

Comment:
I wonder… should the REST API rather be used?

Reply:
Also, maybe it is premature to create an abstract `AjaxAction` construct. How about just naming this to be specific for the purpose of the re-enabling transient caching? Once we have another use case, then it would make sense to create an abstraction construct since we'll have another data point to inform it.